numba/cpython/setobj.py

```diff
@@ -1351,24 +1406,60 @@ def set_difference_update(context, builder, sig, args):
 
     return context.get_dummy_value()
 
-@lower_builtin("set.intersection_update", types.Set, types.Set)
+
+@intrinsic
+def _set_difference_update(typingctx, a, b):
+    sig = types.none(a, b)
+    return sig, set_difference_update
+
+
+@overload_method(types.Set, "difference_update")
+def set_difference_update_impl(a, b):
+    if not all([isinstance(typ, types.Set) for typ in (a, b)]):
+        return
```

Comment:
Factor this out as function `all_sets(*args)` ?

Reply:
Also, the original `_resolve_xxx_update` method requires that the `dtype` of the two sets is the same:
https://github.com/numba/numba/blob/4bc04ceb28a3bce0461e23db16b8826745e96aef/numba/core/typing/setdecl.py#L92-L98